lib/run_loop/detect_aut/xcode.rb

```diff
@@ -147,6 +162,23 @@ def xcode_preferences_plist
       def pbuddy
         @pbuddy ||= RunLoop::PlistBuddy.new
       end
+
+      # @!visibility private
+      def self.detect_selected_device
+        file_name = find_user_state_file
+        pbuddy ||= RunLoop::PlistBuddy.new
```

Comment:
This: `||=` is incorrect in this case.  The method-local variable `pbuddy` is _always_ nil.